content/blog/run_deepseek_locally.md

```diff
@@ -0,0 +1,229 @@
+- **Choose the right size** - From small models for basic computers to larger models for advanced tasks
+
+## Getting Started in 3 Simple Steps
+
+### Step 1: Install Ollama
+
+1. **Download the installer**:
+   - Visit [ollama.com/download](https://ollama.com/download) 
+   - Download the version for your operating system (Windows, macOS, or Linux)
+   - Follow the installation instructions
+
+2. **Verify installation**:
+   ```bash
+   ollama --version  # Should show the current version number
+   ```
+   {{< image "run_deepseek_locally/ollama_version.webp" "Verify Ollama Installation" >}}
+
+### Step 2: Download a DeepSeek-R1 Model
+
+DeepSeek-R1 comes in different sizes to match your computer's capabilities:
+
+1. **For basic computers** (8GB RAM or less):
+   ```bash
+   ollama pull deepseek-r1:1.5b
+   ```
+   {{< image "run_deepseek_locally/deepseek_r1_1.5b_pull.webp" "Pulling the smallest DeepSeek-R1 model" >}}
+
+2. **For mid-range computers** (16GB RAM):
+   ```bash
+   ollama pull deepseek-r1:7b
+   ```
+
+3. **For powerful computers** (32GB+ RAM):
+   ```bash
+   ollama pull deepseek-r1:8b    # 4.9GB
+   ollama pull deepseek-r1:14b   # 9.0GB
+   ```
+
+4. **Check your downloaded models**:
+   ```bash
+   ollama list
+   ```
+   {{< image "run_deepseek_locally/ollama_list.webp" "List of your available models" >}}
+
+### Step 3: Start Using DeepSeek-R1
+
+Now that you have the model downloaded, let's start using it:
+
+1. **Run the model**:
+   ```bash
+   ollama run deepseek-r1:1.5b
+   ```
+   {{< image "run_deepseek_locally/deepseek_r1_run.webp" "DeepSeek-R1 running in terminal" >}}
+
+2. **Try reasoning tasks**:
+   ```
+   You: Solve this step by step: If I have 12 apples and give away 1/3 of them, then eat 2, how many do I have left?
+   ```
+   {{< image "run_deepseek_locally/command_output_1.webp" "DeepSeek-R1 Try reasoning tasks1" >}}
+   {{< image "run_deepseek_locally/command_output_2.webp" "DeepSeek-R1 Try reasoning tasks2" >}}
+
+## Using the API for Your Applications
+
+You can also use DeepSeek-R1 in your applications through the Ollama API:
+
+1. **Make sure the server is running**:
+   ```bash
+   ollama serve
+   ```
+
+2. **Send requests with curl**:
+   ```bash
+   curl --location 'http://localhost:11434/api/chat' \
+   --header 'Content-Type: application/json' \
+   --data '{
+     "model": "deepseek-r1:1.5b",
+     "messages": [
+       {
+         "role": "user",
+         "content": "Hello"
+       }
+     ]
+   }'
+   ```
+{{< image "run_deepseek_locally/postman_ss.webp" "Send requests with curl" >}}
+
+3. **To test the Ollama API using JavaScript, follow these simple steps**:
+    -  Clone the repository from GitHub: {{< link href="https://github.com/Moksh45/RunOllamaApi" >}}RunOllamaApi{{< /link >}}.
+    - Open the project directory in your terminal.
+    - Install the required dependencies by running `npm install`.
+    - Execute the script with `node main.js` to test the API.
+
```

Comment:
Add a section to Use it through GUI

Write how to use open web ui and add a screenshot.

Reply:
Done